src/sas/qtgui/Utilities/OrientationViewer/OrientationViewer.py

```diff
@@ -0,0 +1,314 @@
+from typing import Optional, List
+import numpy as np
+
+from PyQt5 import QtWidgets
+from PyQt5.QtWidgets import QSizePolicy
+from PyQt5.QtCore import Qt
+
+import matplotlib as mpl
+
+from sasmodels.core import load_model_info, build_model
+from sasmodels.data import empty_data2D
+from sasmodels.direct_model import DirectModel
+from sasmodels.jitter import Rx, Ry, Rz
+
+from sas.qtgui.GL.color import Color
+from sas.qtgui.GL.scene import Scene
+from sas.qtgui.GL.transforms import Rotation, Scaling, Translation
+from sas.qtgui.GL.surface import Surface
+from sas.qtgui.GL.cylinder import Cylinder
+from sas.qtgui.GL.cone import Cone
+from sas.qtgui.GL.cube import Cube
+
+from sas.qtgui.Utilities.OrientationViewer.OrientationViewerController import OrientationViewierController, Orientation
+
+
+def createCubeTransform(theta_deg: float, phi_deg: float, psi_deg: float, scaling: List[float]) -> np.ndarray:
+    # Get rotation matrix
+    r_mat = Rz(phi_deg) @ Ry(theta_deg) @ Rz(psi_deg) @ np.diag(scaling)
```

Comment:
That's what it is in the original, seems right to me. Remember that it rotates the whole coordinate frame, so that the `Rz` after `Ry` is different from the first.

Reply:
See wikipedia article on [Euler angles](https://en.wikipedia.org/wiki/Euler_angles#Signs,_ranges_and_conventions): first and third rotation are about the same axis.